src/tarball.py

```diff
@@ -56,8 +56,20 @@ def _find_build_config(self, node):
 
     def _update_repo(self, config):
         self.log.info(f"Updating repo for {config.name}")
-        kernelci.build.update_repo(config, self._kdir)
+        try:
+            kernelci.build.update_repo(config, self._kdir)
+        except Exception as err:
+            self.log.error(f"Failed to update repo: {err}")
+            # safeguard, make sure it is git repo
+            if not os.path.exists(os.path.join(self._kdir, '.git')):
+                self.log.error(f"{self._kdir} is not a git repo")
+                return
+            # cleanup the repo and return False, so we try again
+            kernelci.shell_cmd(f"rm -rf {self._kdir}")
```

Comment:
Although I haven't tested it I think this will raise permission issues.

Reply:
Its already tested on staging, as @pawiecz was guessing why there is no results over weekend, we had by some reason git.lock problem popped up again, so it was perfect opportunity to test this patch.
Verifying .git is just safeguard, to not rm -rf something important.

Reply:
Okay. That's great.